### ISSUE TITLE
Validation Logs

### ISSUE THREAD

**Issue by Shai2022** [gmberton/CosPlace#42]
Could you please provide validation logs for the  training using ResNet101/152 or any other ResNet?
Thanks!

**Comment by gmberton** [comment]
Hi, here are the logs for ResNet-101 and ResNet-152 with output dimensionality 2048.
Note that (1) each of these is actually two logs concatenated, as I had to do resume while training, (2) these were run with an old GTX 1070 GPU so the experiments were quite slow and (3) they are computed with an older private version of the code, so some of the logs are in Italian instead of English, and some other small differences might exist between these and logs obtained with the open-source code from this repo.

Let me know if you need other logs :-)

[ResNet-101-2048-info.log](https://github.com/gmberton/CosPlace/files/14314394/ResNet-101-2048-info.log)
[ResNet-152-2048-info.log](https://github.com/gmberton/CosPlace/files/14314395/ResNet-152-2048-info.log)

**Comment by Shai2022** [comment]
thanks! 
Also, is the validation in the small dataset and the full dataset the same?

**Comment by gmberton** [comment]
Yes, they are the same

**Comment by Shai2022** [comment]
Thanks !
I started training with ResNet152, but the results do not match your logs.
I have included the logs here, and I would appreciate your help.

Namespace(L=2, M=10, N=5, alpha=30, augmentation_device='cuda', backbone='ResNet152',
 batch_size=32, brightness=0.7, classifiers_lr=0.01, contrast=0.7, device='cuda', epochs_num=50, fc_output_dim=2048, g
roups_num=8, hue=0.5, image_size=512, infer_batch_size=16, iterations_per_epoch=10000, lr=1e-05, min_images_per_class=
10, num_preds_to_save=0, num_workers=8, output_folder='logs/default/2024-02-16_19-13-51', positive_dist_threshold=25, 
random_resized_crop=0.5, resize_test_imgs=False, resume_model=None, resume_train=None, saturation=0.7, save_dir='defau
lt', save_only_wrong_preds=False, seed=0,
2024-02-16 19:13:51   The outputs are being saved in logs/default/2024-02-16_19-13-51                                 
2024-02-16 19:13:55   Train only layer3 and layer4 of the ResNet152, freeze the previous ones                         
2024-02-16 19:13:55   There are 4 GPUs and 96 CPUs.                                                                   
2024-02-16 19:13:56   Using cached dataset cache/_M10_N5_alpha30_L2_mipc10.torch                                      
2024-02-16 19:14:22   Using 8 groups                                                                                  2024-02-16 19:14:22   The 8 groups have respectively the following number of classes [35790, 35922, 35214, 35526, 3595
8, 35046, 35520, 35610]
2024-02-16 19:14:22   The 8 groups have respectively the following number of images [706128, 709044, 688956, 702792, 7
12152, 695616, 689178, 703554]
2024-02-16 19:14:22   There are 35790 classes for the first group, each epoch has 10000 iterations with batch_size 32,
 therefore the model sees each class (on average) 8.9 times per epoch
100%|█████████████████████████████████████████████████████████| 10000/10000 [44:37<00:00,  3.73it/s]
2024-02-16 19:59:00   Epoch 00 in 0:44:38, loss = 15.9895
2024-02-16 19:59:00   Extracting database descriptors for evaluation/testing
100%|█████████████████████████████████████████████████████████████| 501/501 [00:35<00:00, 14.09it/s]
2024-02-16 19:59:36   Extracting queries descriptors for evaluation/testing using batch size 1
100%|███████████████████████████████████████████████████████████| 7993/7993 [02:36<00:00, 51.00it/s]
2024-02-16 20:02:13   Calculating recalls
2024-02-16 20:02:16   Epoch 00 in 0:47:53, < val - #q: 7993; #db: 8015 >: R@1: 3.2, R@5: 6.9,

**Comment by gmberton** [comment]
Are you using multiple GPUs to train? I think that might not work.
And did you make any changes to the code?
Also, perhaps it's faster to try a ResNet-18 to make sure that it works fine, at least for one epoch.

**Comment by Shai2022** [comment]
i am using only one gpu for the trainig.
this is the result of resnet18:
Epoch 00 in 0:08:13, < val - #q: 7993; #db: 8015 >: R@1: 17.1, R@5: 24.6
is it to low?

**Comment by gmberton** [comment]
It's very low, it should be around `R@1: 74.2` at the first epoch.
Did you change something in the code?

**Comment by Shai2022** [comment]
I found the problem
thank you!
